File: android/src/androidTest/java/org/ligi/gobandroid_hd/base/GobandroidTestBaseUtil.java
package org.ligi.gobandroid_hd.base;

import android.content.Context;
import android.content.res.AssetManager;

import org.ligi.axt.AXT;

import java.io.IOException;
import java.io.InputStream;

import static org.assertj.core.api.Fail.fail;

public class GobandroidTestBaseUtil {
    protected static String readAsset(final Context context, final String fileName) {
        try {
            final AssetManager assets = context.getAssets();
            final InputStream inputStream = assets.open(fileName);
            return AXT.at(inputStream).readToString();
        } catch (IOException e) {
            fail("could not read test asset " + fileName + e);
            return null;
        }
    }
}

File: android/src/androidTest/java/org/ligi/gobandroid_hd/test_helper_functions/GobandroidTestBaseUtil.kt
package org.ligi.gobandroid_hd.test_helper_functions

import android.content.Context
import org.assertj.core.api.Fail.fail
import java.io.IOException
import java.io.StringWriter

fun readAssetHowItShouldBe(context: Context, fileName: String): String? {
    try {
        val assets = context.assets
        val inputStream = assets.open(fileName)
        return inputStream.reader().readText()
    } catch (e: IOException) {
        fail("could not read test asset $fileName $e")
        return null
    }
}

fun readAsset(context: Context, fileName: String): String? {

        val assets = context.assets
        val inputStream = assets.open(fileName)
        val foo = StringWriter()
        inputStream.reader().copyTo(foo)
        return foo.toString()


}
